File: InTouch/app/src/main/java/com/n8/intouch/MainActivity.java
package com.n8.intouch;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

import com.firebase.client.Firebase;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Firebase firebase = ((InTouchApplication) getApplication()).getFirebase();

        Toast.makeText(this, firebase.getKey(), Toast.LENGTH_LONG).show();

        firebase.child("foo").setValue("goobaroo");
    }
}

File: InTouch/app/src/main/java/com/n8/intouch/MainActivity.kt
package com.n8.intouch

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

import com.firebase.client.Firebase
import com.n8.intouch.dependency_injection.DaggerApplicationComponent
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var firebase: Firebase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        InTouchApplication.graph.inject(this)

        setContentView(R.layout.activity_main)

        firebase.child("foo").setValue("goobaroo")
    }
}
